x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RealmsAuthenticator.java

```diff
@@ -279,15 +279,19 @@ public void lookupRunAsUser(Context context, Authentication authentication, Acti
         assert authentication.getLookedUpBy() == null : "authentication already has a lookup realm";
         final String runAsUsername = context.getThreadContext().getHeader(AuthenticationServiceField.RUN_AS_USER_HEADER);
         if (runAsUsername != null && runAsUsername.isEmpty() == false) {
-            logger.trace("Looking up run-as user [{}] for authenticated user [{}]", runAsUsername, authentication.getUser().principal());
+            logger.trace(
+                "Looking up run-as user [{}] for authenticated user [{}]",
+                runAsUsername,
+                authentication.getEffectiveSubject().getUser().principal()
```

Comment:
Should we use `getAuthenticatingSubject` here instead? I know at this stage in the flow that will be the same as `getEffectiveSubject` but it's slightly more accurate. No need to do it as part of this PR since it may have cascading changes to tests where we mock

Reply:
Good catch! I'll update this and below to use `getAuthenticatingSubject`. Will fix the tests if there are any failure. Thanks!